### PR TITLE
Improve test coverage for /pkg/controllers/certificate, with a minor optimization

### PR DESCRIPTION
**Pull request by xovoxy**
**What type of PR is this?**
/kind cleanup
/kind failing-test

**What this PR does / why we need it**:
Add the test for  /pkg/controllers/certificate, improve test coverage to 55%; 
found some places in `CertRotationController` did not use a unified context,  changed it together.
**Which issue(s) this PR fixes**:
Part of #https://github.com/karmada-io/karmada/issues/5235

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @xovoxy thanks for your contribution.
There are some e2e fixes that have been incorporated recently. Can you rebase the recent code in the master branch so that CI should pass?

**Comment by xovoxy** [comment]
> Hi @xovoxy thanks for your contribution. There are some e2e fixes that have been incorporated recently. Can you rebase the recent code in the master branch so that CI should pass?

Thanks for your tips ~

**Comment by XiShanYongYe-Chang** [comment]
Hi @xovoxy would you like to review the ut tests submitted by others? It's not mandatory, you can do it as you wish, and it's an important contribution to the community. Thank you very much! (I highly encourage you to review each other. My perspective may be subjective, but you can discuss it fully and provide more valuable review comments.)

**Comment by xovoxy** [comment]
> Hi @xovoxy would you like to review the ut tests submitted by others? It's not mandatory, you can do it as you wish, and it's an important contribution to the community. Thank you very much! (I highly encourage you to review each other. My perspective may be subjective, but you can discuss it fully and provide more valuable review comments.)

Of course, this is also a good experience for me. Thank you for your invitation~

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Ask @zhzhuang-zju help take a review.
/cc @zhzhuang-zju

File: pkg/controllers/certificate/cert_rotation_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
...RUZBQU9DQVE4QU1JSUJDZ0tDQVFFQXFWSGZiUUVxdHVhdHYxU3IKeEtXaTZrVTJpa0FDdTA4eFJJK2JCU1ZyNjZkbEpZYklVb2ozR0hoRTlXV3h0bVlTRjVjNU90eGJraVRiWVcyeAoxRjZ5RHAzS21SRE9VdXJreGFBSDFjNW5wbFFrYy9mM1lFVGpuTlpKUGtvTjhnWkpwclpud2FaU0lWNXppWW9ECm5rKzZ6NjJEM25pQ1ZCNjlCaDdVN0xNMXZlOGRIQXZPc0lJVUNKZ3hQVFVobUZVMEswTEtHQWZJamJHRS85RU4Kd1VhVkpVNGYvQi9aL0ppS20yaU1WMmJWdTVTUG5ubThNYjZXSGNuclVOY3Bpdkw4U0pCdG1aS0V3Z3ZGWkhJZgpMSVJBZmNmYmR5b0w0d2R5dFd3ZHNhcEpXd3A3NG5yRURudGFLSVBjYkFvVmNUSldacmtnaWRnVHFqcTN0WS85CkZwMmNtUUlEQVFBQm8xWXdWREFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJS3dZQkJRVUgKQXdJd0RBWURWUjBUQVFIL0JBSXdBREFmQmdOVkhTTUVHREFXZ0JReUJVbXNXT08rV1ZXVFdjTFlLZlRLOFQ1RQpuekFOQmdrcWhraUc5dzBCQVFzRkFBT0NBUUVBSGQ5YU92Q0pRcFBYdzdQSmE2bSs3ZUNYSmxUdy9lZXFhT1ptCkUvU3F5MER5U0hLalptZWR4WjFHdTlEL2Q0Qnl6Ri82VUFGTjArSjZ5bUdZa2Y3TzNqempCSmh6a2U1UzRzbHMKWUpZU290cWd1VW5YakpCTVJNZzl0YUpSY1RiSU9oVmNHdDByd2t2NDB6V3JpdEplTTFEYmV0Q3FORm9xcms0bQo1WWszWHVwT01VejRtb0dnUzVDbmdIS0xXckVMRjI5VWFsM1BXRUdRbnZoVEc0N2dtUmk2dVVTUGNreGJnVm4wCithVU5Rei9Yc0tpOXgwUEtQU2xJTzMrMFVaMTdIa3Zlb1ZHOGdXRkR5bU94ckMzcEZHUkNNWTFJM3RVdUtvMTYKVUI5NW1XSFFUNjlBZllyanh3R3J4QzdkNTRKL1Y1NERnSUF6cVhFT2FFK1pKTHVYTFE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
```
Comment (zhzhuang-zju) [inline pkg/controllers/certificate/cert_rotation_controller_test.go:144]: The certificate will expire on August 15 next year, which may cause this use case to fail. Can you provide a certificate with a longer validity period, and add comments to indicate the expiration time?

Comment (zhzhuang-zju) [reply pkg/controllers/certificate/cert_rotation_controller_test.go:144]: Is this test case necessary, considering it is effectively covered by `func Test_getKubeconfigFromSecret` and `func TestCertRotationController_shouldRotateCert`?

Comment (xovoxy) [reply pkg/controllers/certificate/cert_rotation_controller_test.go:144]: oh yes, thinks for you suggestion~    let me handle it

**Comment by xovoxy** [comment]
hi @XiShanYongYe-Chang @zhzhuang-zju, I’ve updated the issue mentioned above. Could you please review it again? Thanks～

**Comment by XiShanYongYe-Chang** [comment]
Kindly ping @zhzhuang-zju, can you help take a review again?

**Comment by zhzhuang-zju** [comment]
> Kindly ping @zhzhuang-zju, can you help take a review again?

sure

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/approve

**Pull request merged**